Fix pods not scampering (#1008)

pledbrook opened the pull request:
The `SkipTurn()` implementation in `XGAIBehavior` skips the entire pod's turn if the pod starts the turn "unrevealed", which basically means they haven't activated by the start of the turn. This happens when the pod
leader skips moving on activation, for example if they're immobilized or panicked.

The fix I've implemented here is to check whether the pod is scampering and only skip the pod's turn if they *aren't*. This ensures that the pod members that can scamper actually do so.

Fixes #1008.

Iridar reviewed (approved):
Solution make sense, needs (public) testing to make sure it catches what it is supposed to catch.

pledbrook commented:
Tested with the config var. Behaves as expected.

Iridar commented:
> Tested with the config var. Behaves as expected.

Ah good. Question; why is this gated behind a config var? Seems like a straightforward bugfix to me.

pledbrook commented:
Config var was part of a potential compromise to get the change into 1.22. That didn't happen.

Iridar commented:
> Config var was part of a potential compromise to get the change into 1.22. That didn't happen.

I see. Personally I think it's fine to include this into 1.22.2 along with #1032, without the config var, enabled as baseline. @Xymanek, what's your take on this?

Xymanek commented:
I feel like we keep going in circles on this matter. My position was that 1.22.0 has a lot of stuff already and if something went wrong (which did happen) the debugging process would painful due to the huge diff. I suggested that we wait until 1.22.0 (well, 1.22.2 at this point) is released to stable, then ~2 week(end)s and then if we don't get any reports of issues, we can start preparing 1.23, including this. Nothing has changed, I still stand by this opinion.

P.S. I do agree that this should be done without the config var

Iridar commented:
> I feel like we keep going in circles on this matter. My position was that 1.22.0 has a lot of stuff already and if something went wrong (which did happen) the debugging process would painful due to the huge diff. I suggested that we wait until 1.22.0 (well, 1.22.2 at this point) is released to stable, then ~2 week(end)s and then if we don't get any reports of issues, we can start preparing 1.23, including this. Nothing has changed, I still stand by this opinion.
> 
> P.S. I do agree that this should be done without the config var

Fine be me. My memory is trash and the last week was super busy, so I was a bit fuzzy on the matter, sorry.

Iridar commented:
In that case @pledbrook please update this PR to go without a config var and we'll merge it into 1.23 when the time comes.

pledbrook commented:
Removed config variable and rebased against `master`. No reports from LWOTC of issues related to this change and it appears to fix the issues with pods not scampering.
